CMakeLists.txt

```diff
@@ -12,6 +12,12 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
+# We need to tell cmake where the function for the compiler warnings is
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+include(CompilerWarnings)
+
+set_project_warnings()
+
 # We use additional modules that cmake needs to know about
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/Modules/")
```

Comment:
Lines 15 and 22 could be combined

Reply:
`option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" TRUE)` should be added somewhere as well